src/lib/core/CHIPTLVDebug.cpp

```diff
@@ -147,7 +147,10 @@ static void DumpHandler(DumpWriter aWriter, const char * aIndent, const TLVReade
         case kTLVType_ByteString:
             err = temp.GetDataPtr(strbuf);
             VerifyOrExit(err == CHIP_NO_ERROR, aWriter("Error in kTLVType_ByteString"));
-            aWriter("%p\n", strbuf);
+            for (uint32_t i = 0; i < len; i++)
+            {
+                aWriter("%02X", strbuf[i]);
```

Comment:
```suggestion
                aWriter("0x%02X ", strbuf[i]);
```

Reply:
This makes it more obvious it is not decimal, and prevents confusion around endian-ness.

Reply:
Adopted format proposed by @tcarmelveilleux : "hex:"